pkg/ring/lifecycler.go

```diff
@@ -429,6 +451,8 @@ func (i *Lifecycler) autoJoin(ctx context.Context) error {
 		sort.Sort(sortableUint32(tokens))
 		i.setTokens(tokens)
 
+		i.updateCounters(ringDesc)
```

Comment:
Right I don't think we should update the counter here, the ring updation might or might not have succeeded.

Reply:
It should be done in `loop()`.

Reply:
I've followed a different path instead of doing it in the `loop()`, to keep the loop clear more clean. May you check it out, please?